### PR TITLE
feat: [CDS-44558]: migrating app defaults

### PR DESCRIPTION
**Pull request by prabu-kr**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38754)

### CODE REVIEW

File: 100-migrator/src/main/java/io/harness/ngmigration/service/entity/AppMigrationService.java [hunk 1]
```diff
@@ -110,9 +138,59 @@ public NGMigrationStatus canMigrate(NGMigrationEntity entity) {
   }
 
   @Override
+  public MigrationImportSummaryDTO migrate(String auth, NGClient ngClient, PmsClient pmsClient,
+                                           MigrationInputDTO inputDTO, NGYamlFile yamlFile) throws IOException {
+    Application application = appService.getApplicationWithDefaults(yamlFile.getCgBasicInfo().getAppId());
```
Comment (puthrayaharness) [inline 100-migrator/src/main/java/io/harness/ngmigration/service/entity/AppMigrationService.java:143]: Don't do this. Do this at the time of discovery itself.

File: 100-migrator/src/main/java/io/harness/ngmigration/service/entity/AppMigrationService.java [hunk 2]
```diff
@@ -110,9 +138,59 @@ public NGMigrationStatus canMigrate(NGMigrationEntity entity) {
...
+
+    if(EmptyPredicate.isEmpty(application.getDefaults())){
+      return MigrationImportSummaryDTO.builder().success(true).errors(Collections.emptyList()).build();
+    }
+
+    String projectIdentifier = MigratorUtility.getProjectIdentifier(PROJECT, inputDTO);
+    String orgIdentifier = MigratorUtility.getOrgIdentifier(PROJECT, inputDTO);
+
+    if(projectIdentifier == null || orgIdentifier == null){
```
Comment (puthrayaharness) [inline 100-migrator/src/main/java/io/harness/ngmigration/service/entity/AppMigrationService.java:152]: MigratorUtility already takes care of throwing exceptions of customers do not provide the details. This check is not needed.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 100-migrator/src/main/java/io/harness/ngmigration/service/entity/AppMigrationService.java (hunk 2 above)
Comment (prabu-kr) [reply 100-migrator/src/main/java/io/harness/ngmigration/service/entity/AppMigrationService.java]: Done

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 2 of 3 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38754)* status: :shipit: complete! ✓

**Pull request merged**